web/src/main/java/org/wso2/testgrid/web/api/ProductService.java

```diff
@@ -138,6 +138,44 @@ public Response getAllProductStatuses() {
         return Response.status(Response.Status.OK).entity(list).build();
     }
 
+    /**
+     * This method returns the product if the requested product name exists in the TestGrid.
+     *
+     * <p> The product is returned as a json response with the last build information and
+     * the last failed build information<p/>
+     *
+     * @return product
+     */
+    @GET
+    @Path("/product-status/{productName}")
+    public Response getProductStatus(
+            @PathParam("productName") String productName) {
+        try {
+            TestPlanUOW testPlanUOW = new TestPlanUOW();
+            ProductUOW productUOW = new ProductUOW();
+            ProductStatus productStatus = new ProductStatus();
+            Optional<Product> productInstance = productUOW.getProduct(productName);
+            Product product;
+            if (productInstance.isPresent()) {
+                product = productInstance.get();
+                productStatus.setProductId(product.getId());
+                productStatus.setProductName(product.getName());
+                productStatus.setLastfailed(APIUtil.getTestPlanBean(testPlanUOW.getLastFailure(product), false));
+                productStatus.setLastBuild(APIUtil.getTestPlanBean(testPlanUOW.getLastBuild(product), false));
+                productStatus.setProductStatus(testPlanUOW.getCurrentStatus(product).toString());
+            } else {
+                String msg = "Could not found a product for requested product name in the TestGrid.";
+                logger.error(msg);
+                return Response.status(Response.Status.NOT_FOUND).entity(msg).build();
+            }
+            return Response.status(Response.Status.OK).entity(productStatus).build();
+        } catch (TestGridDAOException e) {
+            String msg = "Error occurred while fetching the Product statuses ";
+            logger.error(msg, e);
```

Comment:
Same as above. When running multiple products a production system, this error log won't provide enough data to debug an issue.

Reply:
Fixed with 4be247e